File: src/Tests/Performance/PerfSort/PerfSort.go
package main

import (
	"fmt"
	"sort"
	"time"
)

// Sort 2M pseudo-random ints (deterministic xorshift64 fill): measures
// comparison-sort throughput through the sort package abstraction.

func run(n int) int {
	a := make([]int, n)
	var x uint64 = 88172645463325252

	for i := 0; i < n; i++ {
		x ^= x << 13
		x ^= x >> 7
		x ^= x << 17
		a[i] = int(x % 1000000007)
	}

	sort.Ints(a)

	return a[0] + a[n/4] + a[n/2] + a[3*n/4] + a[n-1]
}

func main() {
	start := time.Now().UnixNano()

	total := run(2000000)

	elapsed := time.Now().UnixNano() - start
	fmt.Println("checksum:", total)
	fmt.Println("elapsed_ns:", elapsed)
}

File: src/Tests/Performance/PerfSort/PerfSort.cs
namespace go;

using fmt = fmt_package;
using sort = sort_package;
using time = time_package;

partial class main_package {

internal static nint run(nint n) {
    var a = new slice<nint>(n);
    uint64 x = 88172645463325252UL;
    for (nint i = 0; i < n; i++) {
        x ^= (uint64)((x << (int)(13)));
        x ^= (uint64)((x >> (int)(7)));
        x ^= (uint64)((x << (int)(17)));
        a[i] = (nint)(x % 1000000007);
    }
    sort.Ints(a);
    return a[0] + a[n / 4] + a[n / 2] + a[3 * n / 4] + a[n - 1];
}

internal static void Main() {
    var start = time.Now().UnixNano();
    nint total = run(2000000);
    var elapsed = time.Now().UnixNano() - start;
    fmt.Println("checksum:", total);
    fmt.Println("elapsed_ns:", elapsed);
}

} // end main_package
